test/test_geometries/custom_lon_lat_point.hpp

```diff
@@ -72,14 +72,7 @@ struct access<rw_lon_lat_point<CoordinateType, CoordinateSystem>, Dimension>
     void set(rw_lon_lat_point<CoordinateType, CoordinateSystem>& p,
              CoordinateType const& value)
     {
-        if (Dimension == 0)
-        {
-            p.longitude = value;
-        }
-        else
-        {
-            p.latitude = value;
-        }
+        ( Dimension == 0 ? p.longitude : p.latitude ) = value;
```

Comment:
If by "everywhere" you mean "C++ standard-conformant compiler" then yes :)
The result of conditional operator is an l-value if both the second and third operands are of the same type and both are l-values. In C this operator always returns r-value.

In C++ this operator allows to do many more interesting things. E.g. see the example here: http://en.cppreference.com/w/cpp/language/operator_other#Conditional_operator

I've implemented it this way because it's "compatible" with the implementation of `get()` method, the conditional operator is also used there. It doesn't require any additional utils/macros, the language allows us to do it. And it's only a test.

Though I agree that this is not a well-known feature and a reader may be surprised seing this. So if you strongly prefer the "longer" version with macro I'll change it.